draft/ZEP0006.md

```diff
@@ -0,0 +1,248 @@
+          "title": "Attrs",
+          "type": "object"
+        },
+        "shape": {
+          "title": "Shape",
+          "type": "array",
+          "items": {
+            "type": "integer"
+          }
+        },
+        "chunks": {
+          "title": "Chunks",
+          "type": "array",
+          "items": {
+            "type": "integer"
+          }
+        },
+        "dtype": {
+          "title": "Dtype",
+          "anyOf": [
+            {
+              "type": "string"
+            },
+            {
+              "type": "array",
+              "items": {
+                "type": "string"
+              }
+            }
+          ]
+        },
+        "compressor": {
+          "title": "Compressor",
+          "type": "object"
+        },
+        "fill_value": {
+          "title": "Fill Value"
+        },
+        "order": {
+          "title": "Order",
+          "enum": [
+            "C",
+            "F"
+          ],
+          "type": "string"
+        },
+        "filters": {
+          "title": "Filters",
+          "type": "array",
+          "items": {
+            "type": "object"
+          }
+        },
+        "dimension_separator": {
+          "title": "Dimension Separator",
+          "enum": [
+            ".",
+            "/"
+          ],
+          "type": "string"
+        },
+        "zarr_version": {
+          "title": "Zarr Version",
+          "default": 2,
+          "type": "integer"
+        }
+      },
+      "required": [
+        "attrs",
+        "shape",
+        "chunks",
+        "dtype",
+        "compressor",
+        "order",
+        "filters"
+      ],
+      "additionalProperties": false
+    },
+    "Group": {
+      "title": "Group",
+      "description": "Model of a Zarr Version 2 Group",
+      "type": "object",
+      "properties": {
+        "attrs": {
+          "title": "Attrs",
+          "type": "object"
+        },
+        "members": {
+          "title": "Members",
+          "type": "object",
+          "additionalProperties": {
+            "anyOf": [
+              {
+                "$ref": "#/definitions/Array"
+              },
+              {
+                "$ref": "#/definitions/Group"
+              }
+            ]
+          }
+        },
+        "zarr_version": {
+          "title": "Zarr Version",
+          "default": 2,
+          "type": "integer"
+        }
+      },
+      "required": [
+        "attrs",
+        "members"
+      ],
+      "additionalProperties": false
+    }
+  }
+}
+```
+
+And Zarr V3:
+
+```json
+# insert schema for v3 here
+```
+
+
+## Related Work
+  
+
+
+## Implementation
+
+- pydantic zarr
+- ? 
+
+## Discussion
+
+- todo: show that consolidated metadata can be achieved by applying a flattening transformation to a ZOM representation of a hierarchy.
+- - The origins of consolidated metadata:
+    * <https://github.com/pangeo-data/pangeo/issues/309>
+    * <https://github.com/zarr-developers/zarr-python/pull/268>
+
```

Comment:
it may also be worth summarizing some of the intended benefits to existing/internal applications. For example, the utilization of a standard data object internally within zarr-python may help improve workflow for creating large hierarchies by allowing users to create the ZOM metadata before passing it to a zarr.creation method.